app/scenes/Document/components/CommentThreadItem.tsx

```diff
@@ -107,6 +118,40 @@ function CommentThreadItem({
   const [isEditing, setEditing, setReadOnly] = useBoolean();
   const formRef = React.useRef<HTMLFormElement>(null);
 
+  const handleAddReaction = React.useCallback(
+    async (emoji: string) => {
+      await comment.addReaction({ emoji, userId: user.id });
+    },
+    [comment, user]
+  );
+
+  const handleRemoveReaction = React.useCallback(
+    async (emoji: string) => {
+      await comment.removeReaction({ emoji, userId: user.id });
+    },
+    [comment, user]
+  );
+
+  const fetchReactionData = React.useCallback(async () => {
```

Comment:
Again with the fetch + map, can we just `ReactionList` look after it's own data concerns?

Reply:
The intention is to have `ReactionList` generalized to render on any context &mdash; comment for now, could be used for document later (same as a generalized `ViewReactionsDialog`).